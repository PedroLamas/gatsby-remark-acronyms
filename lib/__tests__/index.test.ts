import Remark from 'remark';
import visit from 'unist-util-visit';

import plugin from '..';

const pluginSettings = {
  acronyms: {
    CSS: 'Cascading Style Sheets',
    HTML: 'Hypertext Markup Language',
  },
};

const remark = Remark().data('settings', {
  commonmark: true,
  footnotes: true,
  pedantic: true,
});

describe('gatsby-remark-acronyms', () => {
  it('renders acronym when one is found on markdown text', () => {
    const markdownAST = remark.parse('I like HTML!');

    const transformed = plugin({ markdownAST }, pluginSettings);

    visit(transformed, 'acronym', node => {
      expect(node.title).toBe(pluginSettings.acronyms.HTML);
    });

    expect(transformed).toMatchSnapshot();
  });

  it('acronyms are case sensitive', () => {
    const markdownAST = remark.parse('I like Html!');

    const transformed = plugin({ markdownAST }, pluginSettings);

    let found = false;

    visit(transformed, 'acronym', () => {
      found = true;
    });

    expect(found).toBeFalsy();

    expect(transformed).toMatchSnapshot();
  });

  it('acronyms are full words', () => {
    const markdownAST = remark.parse('I like XHTML!');

    const transformed = plugin({ markdownAST }, pluginSettings);

    let found = false;

    visit(transformed, 'acronym', () => {
      found = true;
    });

    expect(found).toBeFalsy();

    expect(transformed).toMatchSnapshot();
  });

  it('does not change render if no acronym is found in markdown text', () => {
    const markdownAST = remark.parse('No acronyms on this text!');

    const transformed = plugin({ markdownAST }, pluginSettings);

    let found = false;

    visit(transformed, 'acronym', () => {
      found = true;
    });

    expect(found).toBeFalsy();

    expect(transformed).toMatchSnapshot();
  });
});
