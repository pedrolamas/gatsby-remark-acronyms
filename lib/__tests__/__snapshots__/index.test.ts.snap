// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gatsby-remark-acronyms acronyms are case sensitive 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "type": "text",
          "value": "I like Html!",
        },
      ],
      "position": Object {
        "end": Object {
          "column": 13,
          "line": 1,
          "offset": 12,
        },
        "start": Object {
          "column": 1,
          "line": 1,
          "offset": 0,
        },
      },
      "type": "paragraph",
    },
  ],
  "position": Object {
    "end": Object {
      "column": 13,
      "line": 1,
      "offset": 12,
    },
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "type": "root",
}
`;

exports[`gatsby-remark-acronyms acronyms are full words 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "type": "text",
          "value": "I like XHTML!",
        },
      ],
      "position": Object {
        "end": Object {
          "column": 14,
          "line": 1,
          "offset": 13,
        },
        "start": Object {
          "column": 1,
          "line": 1,
          "offset": 0,
        },
      },
      "type": "paragraph",
    },
  ],
  "position": Object {
    "end": Object {
      "column": 14,
      "line": 1,
      "offset": 13,
    },
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "type": "root",
}
`;

exports[`gatsby-remark-acronyms does not change render if no acronym is found in markdown text 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "type": "text",
          "value": "No acronyms on this text!",
        },
      ],
      "position": Object {
        "end": Object {
          "column": 26,
          "line": 1,
          "offset": 25,
        },
        "start": Object {
          "column": 1,
          "line": 1,
          "offset": 0,
        },
      },
      "type": "paragraph",
    },
  ],
  "position": Object {
    "end": Object {
      "column": 26,
      "line": 1,
      "offset": 25,
    },
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "type": "root",
}
`;

exports[`gatsby-remark-acronyms renders acronym when one is found on markdown text 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "type": "text",
          "value": "I like ",
        },
        Object {
          "type": "html",
          "value": "<abbr title=\\"Hypertext Markup Language\\">HTML</abbr>",
        },
        Object {
          "type": "text",
          "value": "!",
        },
      ],
      "position": Object {
        "end": Object {
          "column": 13,
          "line": 1,
          "offset": 12,
        },
        "start": Object {
          "column": 1,
          "line": 1,
          "offset": 0,
        },
      },
      "type": "paragraph",
    },
  ],
  "position": Object {
    "end": Object {
      "column": 13,
      "line": 1,
      "offset": 12,
    },
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "type": "root",
}
`;
